### DISCUSSION TITLE
Цикл в модальном окне с кнопками вкладок (подчиненных таблиц)

### DISCUSSION THREAD

**Discussion by AlexDeadly** [totumonline/totum-ru-issues-and-discussions#557] [Есть идея, а что если...]
Можно ли как-то открыть цикл так же как он открывается по коду **button_to_cycle**, но в модальном окне?
Сейчас при открытии из таблицы циклов сам цикл выглидит вот так:
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/49980046/3e5308ad-7976-4c66-8b5a-ff588fd7327f)

При этом если в коде **button_to_cycle** сделать вместо `target: 'self'` сделать `target: 'iframe'`, то выглядеть он будет уже вот так:

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/49980046/b8f24244-80b0-4263-b16a-74c95509a262)

То есть никакие кнопоки подчиненных таблиц не отображаются.
Соответственно, при открытии этого же цикла откуда-то из других мест через **linkToTable** в модальном окне тоже нет этих кнопок. Например, при добавлении отгрузки из заказа для того, чтобы указать перечень документов в отгрузке приходится отдельно идти в таблицу всех отгрузок и дописывать данные.

**Comment by ttm-alexeypolunin** [comment]
Добрый. Это есть в списке работ. Сроки неизвестны.

**Reply by erpic-ru** [reply]
Как вариант - можно в модалку добавить нужные кнопки действия/открытия соседних таблиц и, например, через технический префильтр прятать их обычном при переходе из реестра/таблицы циклов

**Reply by AlexDeadly** [reply]
А если сначала открывается реестр отгрузок в модалке, а потом по двойному клику в таблице открывается уже конкретная отгрузка в той же модалке (и вот тут нужны кнопки). Тут как-то можно пробросить факт того, что кнопки надо показать? Может быть есть возможность выяснить, что сейчас это таблица открылась в модальном окне?

**Reply by ttm-alexeypolunin** [reply]
Да, можно. Тк вы открываете через linkToTable, то делаете поле в префильтре скрытое, но доступное к редактированию и в него пробрасываете значение. Например true для чекбокса. И в коде-форматирования таблицы смотрите на это поле в префильтре. Если оно true то не скрываете поля, если false то скрываете лишнее через fielshide.

**Reply by AlexDeadly** [reply]
То есть добавляем скрытый фильтр в таблицу циклов отгрузок. И точно такой же в саму отгрузку.
При переходе из заказа по linkToTable "ставим галочку в скрытом фильтр". В открывшемся модальном окне в списке всех отгрузок в коде button_to_cycle так же при переходе по linkToTable смотрит этот самый скрытый фильтр и его значение пробрасываем глубже в сам цикл уже (в конкретную отгрузку), чтобы показать или скрыть дополнительные кнопки.
Всё верно?

**Reply by ttm-alexeypolunin** [reply]
Да. Это конечно костыль-костыльевичь. Надо добавить как настройку для вкладок, так и в форматированиях переменную определяющую ифрейм или нет.

**Reply by AlexDeadly** [reply]
Проверил. Всё работает.
Из неочень красивого осталось только то, что у модального окна есть кнопка "Вкладка", которая перебрасывает модальное окно в отдельную вкладку и вот там появляются встроенные кнокпи переключения разделов и плюс к этому торчат еще костыльные, так как состояние фильтра остается неизменным.

**Reply by ttm-alexeypolunin** [reply]
Это фикситс параметром bottombottons:fals и topbuttons аналогично ф linkToTable и для dataTable аналогично
